src/frequenz/channels/_receiver.py

```diff
@@ -342,3 +361,79 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         """Return a string representation of the timer."""
         return f"{type(self).__name__}({self._receiver!r}, {self._mapping_function!r})"
+
+
+class _Sentinel:
+    """A sentinel object to represent no value received yet."""
+
+    def __repr__(self) -> str:
+        """Return a string representation of this sentinel."""
+        return "Sentinel()"
```

Comment:
Maybe it is worth adding a `__str__` too in case this is printed somewhere like in `LastValueCache`? I when adding the `__str__` and `__repr__` for `_Filter` it will be used.